csrc/scheduler/cache_policy_refiner.cpp

```diff
@@ -0,0 +1,128 @@
+#include <fusion.h>
+#include <ir/base_nodes.h>
+#include <ir/internal_nodes.h>
+#include <ir/utils.h>
+#include <scheduler/cache_policy_refiner.h>
+#include <scheduler/debug_utils.h>
+
+namespace nvfuser {
+
+// Returns whether a pointwise expression `expr` expands its input operand
+// `in_tv`.
+static bool pointwiseExpands(const Expr* expr, const TensorView* in_tv) {
+  NVF_CHECK(
+      expr->outputs().size() == 1,
+      "A pointwise expression is expected to have one output: ",
+      expr->toString());
+  const Val* out = expr->output(0);
+
+  if (!out->isA<TensorView>()) {
+    return false;
+  }
+  const auto* out_tv = out->as<TensorView>();
+
+  const size_t n_dims = in_tv->getRootDomain().size();
+  if (n_dims != out_tv->getRootDomain().size()) {
+    return false;
+  }
+
+  int num_expands = 0;
+  for (size_t i = 0; i < n_dims; i++) {
```

Comment:
We could also use `PairwiseRootDomainMap`. For example:

```
auto p2c = PairwiseRootDomainMap(in_tv, out_tv).mapBroadcast(true).mapProducerToConsumer(in_tv->domain(), out_tv->domain());
return std::find_if(p2c.begin(), p2c.end(), [](const auto& mapping) {
           mapping.first->isBroadcast() && !mapping.second->isBroadcast();
        } != p2c.end();
}
```

This should work with any producer and consumer pair, so we shouldn't need the filtering of line 66.